app/hotels/src/map/singleHotel/SingleMap.js

```diff
@@ -5,17 +5,26 @@ import { View } from 'react-native';
 import { graphql, createFragmentContainer } from '@kiwicom/mobile-relay';
 import { StyleSheet, StretchedImage } from '@kiwicom/mobile-shared';
 import { defaultTokens } from '@kiwicom/mobile-orbit';
+import {
+  withNavigation,
+  type NavigationType,
+} from '@kiwicom/mobile-navigation';
 
 import MapView from './MapView';
 import AdditionalInfo from './AdditionalInfo';
 import gradient from '../gradient.png';
 import type { SingleMap_hotel as HotelType } from './__generated__/SingleMap_hotel.graphql';
+import CloseModal from '../../components/CloseModal';
 
 type Props = {|
   +hotel: ?HotelType,
+  +navigation: NavigationType,
 |};
 
 const SingleMap = (props: Props) => {
+  function goBack() {
```

Comment:
As far as I understand, this is similar as class construction. Will that be recreated on every render?

Reply:
It probably will. But the future is with react-hooks, and hooks will be available from rn-0.59. This is how we have to create functions with hooks.